ru/turing/courses/lesson3/Registry.java

```diff
@@ -1,13 +1,56 @@
 package ru.turing.courses.lesson3;
 
+import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.Map;
 
-//todo Задание со звездочкой - сделать Registry через generics.
-// Для этого почитать, что такое дженерики и о том, как заставить класс с ними работать
-// Сделать абстрактный доменный класс (похожий на Animal из прошлого урока для этого)
-public class Registry {
-    private final Map<Object, Object> storage = new HashMap<>(); //todo поменять Object, Object на то, что будет у вашего класса в примере
+public class Registry<T extends Number> {
+    private final Map<Integer, T> storage = new HashMap<>();
 
-    //todo написать и реализовать методы
+
+    public void add(Integer key, T value) {
+        this.storage.put(key, value);
+    }
+
+
+    public T removeByKey(Integer key) {
+        T value = this.storage.get(key);
+        this.storage.remove(key);
+        return value;
+    }
+
+
+    public ArrayList removeByValue(T value) {
+        ArrayList<Integer> Keys = new ArrayList<Integer>();
+        for (Map.Entry<Integer, T> entry : this.storage.entrySet()) {
+            if (value == entry.getValue()) {
+                Keys.add(entry.getKey());
+                this.storage.remove(entry.getKey());
+            }
+        }
+        return Keys;
+    }
+
+    public void clear() {
+        this.storage.clear();
+    }
+
+
+    public T getByKey(Integer key) {
+        T value = this.storage.get(key);
+        return value;
+    }
+
+    public ArrayList getByName(T value) {
+        ArrayList<Integer> Keys = new ArrayList<Integer>();
+        for (Map.Entry<Integer, T> entry : this.storage.entrySet()) {
+            if (value == entry.getValue())
+                Keys.add(entry.getKey());
+        }
+        return Keys;
+    }
+
+    public Integer getSize() {
```

Comment:
```suggestion
    public int getSize() {
```
Здесь можно примитив